ERCS/ERC-XXX.md

```diff
@@ -0,0 +1,239 @@
+
+```yaml
+- name: withdrawWithAuth
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: owner
+      type: address
+    - name: assets
+      type: uint256
+    - name: maxShares
+      type: uint256
+    - name: receiver
+      type: address
+    - name: deadline
+      type: uint256
+    - name: auth
+      type: bytes
+
+  outputs:
+    - name: shares
+      type: uint256
+```
+
+#### redeemWithAuth
+
+Burns exactly `shares` from `owner` and sends `assets` of underlying tokens to `receiver` using `owner`'s EIP-3074 AUTH signature, with constraints that ensure the request is processed within the parameters that `owner` is comfortable with.
+
+MUST revert if all of `shares` cannot be withdrawn (due to withdrawal limit being reached, slippage, the owner not having enough shares, etc).
+
+MUST check that the assets redeemed for the withdrawal is at least `minAssets`.
+
+MUST check that the current timestamp is less than or equal to the user setting of `deadline`.
+
+MUST transfer the assets to `receiver`
+
+MUST emit the `Withdraw` event.
+
+MUST return the _exact_ amount of assets transferred to `receiver`.
+
+MUST compute `commit` (per EIP-3074 parlance) as `keccak(b"ERC-<TBD>" || b"redeemWithAuth" || depositor || shares || minAssets || receiver || deadline)`.
+
+MAY support an additional flow in which the shares are transferred to the Vault contract before the `redeemWithAuth` execution, and are accounted for during `redeemWithAuth`.
+
+```yaml
+- name: redeemWithAuth
+  type: function
+  stateMutability: nonpayable
+
+  inputs:
+    - name: owner
+      type: address
+    - name: shares
+      type: uint256
+    - name: minAssets
+      type: uint256
+    - name: receiver
+      type: address
+    - name: deadline
+      type: uint256
+    - name: auth
+      type: bytes
+
+  outputs:
+    - name: assets
+      type: uint256
+```
+
+### Events
+
+`Deposit` and `Withdraw` should be emitted in their corresponding methods, aligning with ERC-4626.
+The event definitions do not change.
+
+## Backwards Compatibility
+
+This extension is fully compatible with ERC-4626.
+The compatibility with other extensions to ERC-20 or ERC-4626 has not been fully analyzed, and may additionally be preferred to leverage this extension to the exclusion of other similar extensions.
+
+## Reference Implementation
+
+TBD
+
+## Security Considerations
+
+All EIP-3074 `commit`s have been pre-pended with two additional string values to ensure replay protection within the context of this contract.
+The number of this ERC has been added to ensure that it does not conflict any other ERCs which may be defined together in the implementation contract.
+The method name of each method called has been additionally added to ensure replayability protection between other methods in this ERC, which have similar order and type to the list of parameters.
+
```

Comment:
```suggestion

Note that all operations defined by this ERC are _NOT_ authenticated _except_ via the EIP-3074 mechanism.
The replay protection defined by this ERC and EIP-3074 is considered enough to prevent a user action from being initated by another party that violates that user's original intent when signing for that action.
In particular, the value of `deadline` MUST be chosen such that it is sufficiently short enough that the user is unlikely to want to be able to perform the same action again, but long enough for that action to succeed (at most, several minutes is suggested, but timing depends on on-chain fee conditions and/or downstream processes such as relayer batch times, etc.)
As a last resort, the EIP-3074 mechanism of performing a transaction to increase the value of that account's `nonce` to reassure the user that a replay is not possible, however this is not a substitute for the proper handling of constraints within the implementation contract.
(Note that for some transactions where the deposit is initiated by the same EOA that has signed for the deposit, there are less concerns due to the change of the user's `nonce` once the action has been processed as a transaction)

```

Reply:
Leaving this unresolved as an important note to read for collaborators